src/Nethermind/Nethermind.Blockchain/ReadOnlyTransactionProcessorFactory.cs

```diff
@@ -0,0 +1,47 @@
+﻿//  Copyright (c) 2018 Demerzel Solutions Limited
+//  This file is part of the Nethermind library.
+// 
+//  The Nethermind library is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU Lesser General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+// 
+//  The Nethermind library is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+//  GNU Lesser General Public License for more details.
+// 
+//  You should have received a copy of the GNU Lesser General Public License
+//  along with the Nethermind. If not, see <http://www.gnu.org/licenses/>.
+
+using Nethermind.Core.Crypto;
+using Nethermind.Core.Specs;
+using Nethermind.Evm;
+using Nethermind.Logging;
+using Nethermind.Store;
+
+namespace Nethermind.Blockchain
+{
+    public class ReadOnlyTransactionProcessorFactory : ITransactionProcessorFactory
+    {
+        private readonly IReadOnlyDbProvider _readOnlyDbProvider;
+        private readonly ISpecProvider _specProvider;
+        private readonly ILogManager _logManager;
+        private readonly ReadOnlyBlockTree _blockTree;
+
+        public ReadOnlyTransactionProcessorFactory(IReadOnlyDbProvider readOnlyDbProvider, IBlockTree blockTree, ISpecProvider specProvider, ILogManager logManager)
```

Comment:
not a great fan of factory pattern in such places, this loses clear composability of dependencies